common/test/acceptance/tests/lms/test_lms_course_discovery.py

```diff
@@ -0,0 +1,86 @@
+"""
+Test course discovery.
+"""
+import datetime
+import json
+import os
+
+from bok_choy.web_app_test import WebAppTest
+from ...pages.common.logout import LogoutPage
+from ...pages.studio.auto_auth import AutoAuthPage
+from ...pages.lms.discovery import CourseDiscoveryPage
+from ...fixtures.course import CourseFixture
+
+
+class CourseDiscoveryTest(WebAppTest):
+    """
+    Test searching for courses.
+    """
+
+    STAFF_USERNAME = "STAFF_TESTER"
+    STAFF_EMAIL = "staff101@example.com"
+    TEST_INDEX_FILENAME = "test_root/index_file.dat"
+
+    def setUp(self):
+        """
+        Create course page and courses to find
+        """
+        # create index file
+        with open(self.TEST_INDEX_FILENAME, "w+") as index_file:
+            json.dump({}, index_file)
+
+        super(CourseDiscoveryTest, self).setUp()
+        self.page = CourseDiscoveryPage(self.browser)
+
+        for i in range(10):
+            org = self.unique_id
+            number = unicode(i)
+            run = "test_run"
+            name = "test course"
+            settings = {'enrollment_start': datetime.datetime(1970, 1, 1).isoformat()}
+            CourseFixture(org, number, run, name, settings=settings).install()
+
+        for i in range(2):
+            org = self.unique_id
+            number = unicode(i)
+            run = "test_run"
+            name = "grass is always greener"
+            CourseFixture(
+                org,
+                number,
+                run,
+                name,
+                settings={
+                    'enrollment_start': datetime.datetime(1970, 1, 1).isoformat()
+                }
+            ).install()
+
+    def tearDown(self):
+        super(CourseDiscoveryTest, self).tearDown()
+        os.remove(self.TEST_INDEX_FILENAME)
```

Comment:
It might be worth using `self.addCleanup()` instead of `tearDown()` to ensure that the directory gets deleted even if an exception occurs in the `setup()` method.